src/realm/realmReducer.js

```diff
@@ -125,11 +126,16 @@ export default (
         crossRealmBots: action.data.cross_realm_bots,
 
         //
-        // InitialDataUpdateDisplaySettings. Deprecated!
+        // InitialDataUserSettings
         //
-        // TODO(#4933): Use modern `user_settings` object for these.
 
-        twentyFourHourTime: action.data.twenty_four_hour_time,
+        twentyFourHourTime:
+          action.data.user_settings?.twenty_four_hour_time
+          // Fall back to InitialDataUpdateDisplaySettings for servers that
+          // don't support the user_settings_object client capability.
+          /* $FlowIgnore[incompatible-cast]: If `user_settings` is absent,
+             this will be present. */
+          ?? (action.data.twenty_four_hour_time: boolean),
```

Comment:
Yeah.

I guess I might've used `$FlowFixMe` if I thought it was reasonably doable to get Flow clued into the "If `user_settings` is absent, this will be present" bit with some clever type definitions. Then this code would be unremarkable and would need no suppression.